lib/command/src/Obelisk/Command/Run.hs

```diff
@@ -69,40 +74,63 @@ data CabalPackageInfo = CabalPackageInfo
+-- | Imports needed to use Obelisk.Run
+obRunImports :: [String]
+obRunImports = [ "import qualified Obelisk.Run"
+               , "import qualified Frontend"
+               , "import qualified Backend" ]
+
+run
+  :: MonadObelisk m
+  => Bool
+  -- ^ Whether to enable profiling in the current project
+  -> m ()
+run profiled = withProjectRoot "." $ \root -> do
   pkgs <- fmap toList . parsePackagesOrFail =<< getLocalPkgs root
-  withGhciScript pkgs root $ \dotGhciPath -> do
-    freePort <- getFreePort
-    assets <- do
-      let importableRoot = toNixPath root
-      isDerivation <- readProcessAndLogStderr Debug $
-        proc nixExePath
-          [ "eval"
-          , "-f"
-          , root
-          , "(let a = import " <> importableRoot <> " {}; in toString (a.reflex.nixpkgs.lib.isDerivation a.passthru.staticFilesImpure))"
-          , "--raw"
-          -- `--raw` is not available with old nix-instantiate. It drops quotation
-          -- marks and trailing newline, so is very convenient for shelling out.
+  freePort <- getFreePort
+  assets <- findProjectAssets root
+  putLog Debug $ "Assets impurely loaded from: " <> assets
+  let obRunExpr = unwords
+          [ "Obelisk.Run.run"
+          , show freePort
+          , "(Obelisk.Run.runServeAsset " ++ show assets ++ ")"
+          , "Backend.backend"
+          , "Frontend.frontend"
           ]
-      -- Check whether the impure static files are a derivation (and so must be built)
-      if isDerivation == "1"
-        then fmap T.strip $ readProcessAndLogStderr Debug $ -- Strip whitespace here because nix-build has no --raw option
-          proc nixBuildExePath
-            [ "--no-out-link"
-            , "-E", "(import " <> importableRoot <> "{}).passthru.staticFilesImpure"
-            ]
-        else readProcessAndLogStderr Debug $
-          proc nixExePath ["eval", "-f", root, "passthru.staticFilesImpure", "--raw"]
-    putLog Debug $ "Assets impurely loaded from: " <> assets
-    runGhcid root True dotGhciPath pkgs $ Just $ unwords
-      [ "Obelisk.Run.run"
-      , show freePort
-      , "(Obelisk.Run.runServeAsset " ++ show assets ++ ")"
-      , "Backend.backend"
-      , "Frontend.frontend"
-      ]
+  case profiled of
+    True -> do
+      putLog Debug "Using profiled build of project."
+      time <- liftIO $ formatTime defaultTimeLocale "%Y-%m-%dT%H:%M:%S" <$> getCurrentTime
+      let exeSource =
+            unlines $
+              [ "module Main where"
+              , "import Control.Exception"
+              , "import Reflex.Profiled" ]
+              <> obRunImports <>
+              [ "main :: IO ()"
+              , "main = " <> obRunExpr <> " `finally` writeProfilingData \"" <> profileBaseName <> ".rprof\"" ]
+          -- Sane flags to enable by default, enable time profiling +
+          -- closure heap profiling.
+          rtsFlags = [ "+RTS", "-p", "-po" <> profileBaseName, "-hc", "-RTS" ]
+          profileDirectory = root </> "profile"
```

Comment:
Seems like using `root` here is a bit odd. I'd expect the output to be in the directory where I ran the command.

Reply:
I would think you'd want it in a predictable place. That way you don't have to remember what directory you were in when you ran it.

By default, the profiling will put it in the cwd of where the executable was run. For things like config/ lookups to work correctly, this has to be the root directory. I guess if we want it to be where `ob profile` was run, we could save the cwd above and put it here.

Reply:
I can very much imagine doing 
```
mkdir test-run1; cd test-run1; ob profile
mkdir test-run2; cd test-run2; ob profile
```

Possibly even in parallel (given the right circumstances).

Reply:
Perhaps best of all would be to take an option like `-o` to actually let me specify instead. That can have a default.

Reply:
Yeah a -o to configure profileBaseName sounds good. At least it makes this use case possible.

The idea with how this is set up is so to pick profileBaseName based on timestamps so multiple runs don't overwrite each other. Ideally this would have more information as well, so you would know what commit you were on, any special actions or notes, and maybe any additional changes from HEAD.